zerver/lib/bugdown/__init__.py

```diff
@@ -882,6 +884,18 @@ def run(self, root: Element) -> None:
                 continue
             vimeo = self.vimeo_image(url)
             if extracted_data:
+                yt_id = self.youtube_id(url)
+                if yt_id is not None:
+                    youtube = self.youtube_image(extracted_data)
+                    yt_title = self.get_youtube_title(extracted_data)
+                    if yt_title is not None:
+                        found_url.family.child.text = yt_title
+                    else:
+                        yt_title = url
+                    add_a(root, youtube, url, yt_title, None,
+                          "youtube-video message_inline_image", yt_id)
+                    continue
```

Comment:
This is a comment that's more about the existing code than your change, but all these `continue` statements are really brittle.  It would be nice to restructure this code some day.  The first step is probably to make a function that handles one video.